src/gui/processing/qgsprocessingaggregatewidgetwrapper.cpp

```diff
@@ -130,12 +130,12 @@ void QgsProcessingAggregatePanelWidget::setValue( const QVariant &value )
   {
     const QVariantMap map = field.toMap();
     const QgsField f( map.value( QStringLiteral( "name" ) ).toString(),
-                      static_cast< QVariant::Type >( map.value( QStringLiteral( "type" ), QVariant::Invalid ).toInt() ),
-                      map.value( QStringLiteral( "type_name" ), QVariant::typeToName( static_cast< QVariant::Type >( map.value( QStringLiteral( "type" ), QVariant::Invalid ).toInt() ) ) ).toString(),
+                      static_cast< QMetaType::Type >( map.value( QStringLiteral( "type" ), QgsVariantUtils::createVariant( QMetaType::Type::UnknownType ) ).toInt() ),
```

Comment:
QVariant [value()](https://doc.qt.io/qt-6/qmap.html#value) takes the default value, meaning a QVariant object in the case of a QVariantMap, so we need a QVariant object. QVariant object was automatically constructed when passing a QVariant::Invalid using this [ctor](https://doc.qt.io/qt-6/qvariant-obsolete.html#QVariant-46) deprecated in Qt 6.

I don't see any other way of fixing this

Reply:
But in this case we're not trying to retrieve a variant from the map, we are trying to retrieve an int/enum value. So I think the better version would be something like:

    static_cast< QMetaType::Type >( map.value( QStringLiteral( "type" ), static_cast< int >( QMetaType::Type::UnknownType ) ).toInt() )

Otherwise the default value will be an invalid QVariant(), we'll call .toInt() on that and get 0 (since QVariant() can't be converted to int, but we aren't checking whether the conversion succeeded), and then static cast it to QMetaType::Type... which still results in QMetaType::Type::UnknownType, but just via a convulated and tricky to understand path :rofl:

Reply:
I'll address this in a follow up, let's get this monster merged before more merge conflicts are introduced!

Reply:
See https://github.com/qgis/QGIS/pull/57620